libsrc/apple2/ser/a2.ssc.s

```diff
@@ -328,21 +330,21 @@ SER_PUT:
 
         ; Try to send
         ldy     SendFreeCnt
-        iny                     ; Y = $FF?
-        beq     :+
+        cpy     #$FF            ; Nothing to flush
+        beq     PutByte
         pha
         lda     #$00            ; TryHard = false
         jsr     TryToSend
         pla
 
-        ; Put byte into send buffer & send
-:       ldy     SendFreeCnt
-        bne     :+
+        ldy     SendFreeCnt     ; Reload
+        bne     PutByte
         lda     #SER_ERR_OVERFLOW
         ldx     #0 ; return value is char
         rts
 
-:       ldy     SendTail
+PutByte:
+        ldy     SendTail
         sta     SendBuf,y
         inc     SendTail
         dec     SendFreeCnt
```

Comment:
Simplification of SER_PUT: cpy and iny are each two cycles, but if we have nothing to send, we can branch directly to PutByte instead of line 339